containers/src/Data/Graph.hs

```diff
@@ -258,7 +265,8 @@ flattenSCCs = concatMap flattenSCC
 -- This function is retained for backward compatibility,
 -- 'flattenSCC1' has the more precise type.
 flattenSCC :: SCC vertex -> [vertex]
-flattenSCC = NE.toList . flattenSCC1
+flattenSCC (AcyclicSCC v) = [v]
+flattenSCC (NECyclicSCC vs) = NE.toList vs
```

Comment:
Ah, so this one won't be too lazy because GHC knows that `NECyclicSCC` is a strict constructor, and `NE.toList` will inline. Subtle. Might be worth a comment.

Reply:
I simply restored the previous definition. But it's probably better to be explicit and add a comment. Done.